process-engine/process_engine_core
# 🐛 Fix race condition on `ExecuteProcessService.startAndAwaitEndEvent`

## Changes

1. Reject EndEventHandler Promise, when the EndEvent is a `TerminateEndEvent`
2. Await both, `executeProcess` and `awaitEndEvent`, before resolving `ExecuteProcessService.startAndAwaitEndEvent`
3. Remove unnecessary `ProcessTerminatedSubscription` from `ExecuteProcessService`
 
## Issues

Part of https://github.com/process-engine/process_engine_runtime/issues/522

PR: #321

## How to test the changes

1)
- Run a Process and let it run into a `TerminateEndEvent`
- Note that the ProcessInstance's final state will be `terminated`

2)
- Run a Process with a suspendable Task (UserTask, ManualTask, etc)
- Terminate it manually
- Note that the ProcessInstance's final state will be `terminated`

3) 
- Run processes through `ExecuteProcessService.startAndAwaitEndEvent` repeatedly
- Note that the request will only resolve, after `executeProcess` has resolved and the `onEndEventReached` notification was received
- Note that the final state transition of the ProcessInstance was done
